apps/rush-lib/src/utilities/Utilities.ts

```diff
@@ -442,6 +443,14 @@ export class Utilities {
     return Utilities._executeLifecycleCommandInternal(command, child_process.spawn, options);
   }
 
+  public static isNonDebugTabCompletionRequest(): boolean {
+    return process.argv.length > 2 && process.argv[2] === CommandLineConstants.TabCompletionActionName;
+  }
+
+  public static shouldPrintBanner(): boolean {
+    return !Utilities.isNonDebugTabCompletionRequest() && process.argv.indexOf('--json') === -1;
```

Comment:
It'd be cool to specifically allow arguments in `ts-command-line` to declare that they signify machine-readable output.

Reply:
`ts-command-line` won't be able to tell dependents until it processes argv though.

Reply:
Let's do this in a separate PR?